jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ExtUtils.java

```diff
@@ -35,10 +65,186 @@ public static String toUri(JarEntryFile jarEntryFile) {
         }
     }
 
+    public static boolean isJarResourceUri(URI uri) {
+        return uri != null && JDT_SCHEME.equals(uri.getScheme()) && CONTENTS_AUTHORITY.equals(uri.getAuthority());
+    }
+
+    public static IJarEntryResource createJarResource(URI uri) throws JavaModelException {
+        String handleId = uri.getQuery();
+        IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) JavaCore.create(handleId);
+        String path = uri.getPath();
+
+        // if the file exists in the java packages
+        String[] segments = StringUtils.split(path, "/");
+        String packageName = StringUtils.join(Arrays.asList(segments).subList(0, segments.length - 1), '.');
+        IPackageFragment packageFragment = packageRoot.getPackageFragment(packageName);
+        if (packageFragment != null && packageFragment.exists()) {
+            Object[] objs = packageFragment.getNonJavaResources();
+            for (Object obj : objs) {
+                if (obj instanceof IJarEntryResource) {
+                    IJarEntryResource child = (IJarEntryResource) obj;
+                    if (child instanceof JarEntryFile && child.getFullPath().toPortableString().equals(path)) {
+                        return child;
+                    }
+                }
+
+            }
+        }
+        Object[] resources = packageRoot.getNonJavaResources();
+
+        for (Object resource : resources) {
+            if (resource instanceof JarEntryFile) {
+                JarEntryFile file = (JarEntryFile) resource;
+                if (file.getFullPath().toPortableString().equals(path)) {
+                    return file;
+                }
+            }
+            if (resource instanceof JarEntryDirectory) {
+                JarEntryDirectory directory = (JarEntryDirectory) resource;
```

Comment:
If the first one failed, it will not continue the search in other folders. Is this by design?

Reply:
good catch